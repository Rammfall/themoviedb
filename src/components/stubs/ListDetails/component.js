import React from 'react'
import {
  Layout, Row, Col, Typography, Modal, Pagination
} from 'antd'
import { MinusCircleOutlined, DeleteOutlined } from '@ant-design/icons'
import { range } from 'lodash'

import Header from '../Header'
import CreateListModal from '../CreateListModal'
import Movie from '../MovieItem'

const showDeleteMovieModal = () => {
  Modal.confirm({
    title: 'Do you want to delete movie from this list?',
    onOk() {},
    onCancel() {}
  })
}

const showDeleteListModal = () => {
  Modal.confirm({
    title: 'Do you want to delete list?',
    onOk() {},
    onCancel() {}
  })
}

class ListDetails extends React.Component {
  constructor(props) {
    super(props)

    this.state = { modalVisible: false }

    this.showModal = () => {
      this.setState({ modalVisible: true })
    }

    this.hideModal = () => {
      this.setState({ modalVisible: false })
    }
  }

  render() {
    const { modalVisible } = this.state

    return (
      <Layout>
        <Header />
        <Layout.Content>
          <Row>
            <Col
              offset={2}
              span={20}
            >
              <div className="top-margin">
                <Typography.Title>
                  List item 1
                  {' '}
                  <MinusCircleOutlined onClick={showDeleteListModal} />
                </Typography.Title>
              </div>
            </Col>
          </Row>
          <Row
            gutter={8}
            type="flex"
          >
            <Col
              span={20}
              offset={2}
            >
              <Row
                gutter={{
                  xs: 8, sm: 16, md: 24, lg: 32
                }}
              >
                {range(10).map(item => (
                  <Col
                    key={item}
                    xs={{ span: 24 }}
                    sm={{ span: 12 }}
                    md={{ span: 8 }}
                    lg={{ span: 8 }}
                    xl={{ span: 6 }}
                  >
                    <Movie
                      actions={[
                        <DeleteOutlined
                          key="delete"
                          onClick={showDeleteMovieModal}
                        />
                      ]}
                    />
                  </Col>
                ))}
              </Row>
            </Col>
          </Row>
          <Row
            type="flex"
            justify="center"
          >
            <Col>
              <Pagination
                defaultCurrent={1}
                total={50}
                className="pagination"
              />
            </Col>
          </Row>
        </Layout.Content>
        <Modal
          visible={modalVisible}
          onCancel={this.hideModal}
          okText="Create"
          title="Create list"
        >
          <CreateListModal />
        </Modal>
      </Layout>
    )
  }
}

export default ListDetails
