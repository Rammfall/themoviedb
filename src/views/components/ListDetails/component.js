import React from 'react'
import PropTypes from 'prop-types'

import PrivateLayout from 'Views/layouts/PrivateLayout'
import Loader from 'Views/components/Loader'
import ListContent from './ListContent'

const ListDetails = ({ isBlank }) => (
  <PrivateLayout>
    {isBlank ? (<Loader />) : <ListContent />}
  </PrivateLayout>
)

ListDetails.propTypes = {
  isBlank: PropTypes.bool
}

ListDetails.defaultProps = {
  isBlank: true
}

export default ListDetails
