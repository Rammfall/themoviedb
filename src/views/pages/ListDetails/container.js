import React, { Component } from 'react'
import { compose } from 'ramda'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import ListDetails from 'Views/components/ListDetails'
import { getListDetails } from 'Store/concepts/listDetails/actions'
import { isBlankSelector } from 'Store/concepts/listDetails/selectors'

class ListDetailsPage extends Component {
  componentDidMount() {
    this.getData()
  }

  componentDidUpdate(prevProps) {
    const { location: { search } } = this.props

    if (search !== prevProps.location.search) {
      this.getData()
    }
  }

  getData = () => {
    const { getMovies, match: { params: { id } } } = this.props

    getMovies({ id })
  }

  render() {
    const { isBlank } = this.props

    return(
      <ListDetails
        isBlank={isBlank}
      />
    )
  }
}

ListDetailsPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  getMovies: PropTypes.func.isRequired,
  isBlank: PropTypes.bool,
  location: PropTypes.shape({
    search: PropTypes.string.isRequired
  }).isRequired
}
ListDetailsPage.defaultProps = {
  isBlank: true
}

const mapStateToProps = (state, { match: { params: { id } } }) => ({
  isBlank: isBlankSelector(state, id)
})

const mapDispatchToProps = {
  getMovies: getListDetails
}

export { ListDetailsPage as ListDetailsPageContainer }
export default compose(
  withRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(ListDetailsPage)
