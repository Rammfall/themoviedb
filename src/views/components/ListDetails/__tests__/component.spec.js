import React from 'react'
import { shallow } from 'enzyme'

import ListDetails from '../component'

describe('ListDetails', () => {
  describe('when isBlank true', () => {
    it('matches snapshot', () => {
      const component = shallow(<ListDetails isBlank />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('when isBlank false', () => {
    it('matches snapshot', () => {
      const component = shallow(<ListDetails isBlank={false} />)

      expect(component).toMatchSnapshot()
    })
  })
})
