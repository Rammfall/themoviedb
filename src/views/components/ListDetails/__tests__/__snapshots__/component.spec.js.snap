// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDetails when isBlank false matches snapshot 1`] = `
<PrivateLayout>
  <withRouter(Connect(injectIntl(ListContent))) />
</PrivateLayout>
`;

exports[`ListDetails when isBlank true matches snapshot 1`] = `
<PrivateLayout>
  <Loader />
</PrivateLayout>
`;
